eth_utils/hexidecimal.py

```diff
@@ -67,5 +67,9 @@ def is_hex(value):
         return False
     except TypeError:
         return False
+    except UnicodeEncodeError:
```

Comment:
Maybe you can comment these to explain how each gets raised?  or add a type check at the beginning of the function to avoid these in the first place?

Reply:
I took a look at type-checking first. Between handling py2 vs py3 and bytes vs unicode vs str, I think it's cleaner to just catch the exception. I'll add a note about what triggers it.

Reply:
Yeah, this is a border case for "Easier to ask for forgiveness than permission" vs "Look before you leap". It's clearer to see what's going on in the pre-test, but it also leaves room for other errors we might not catch (or is maybe too aggressive?).  I'm ambivalent about it. Just pushed a pre-test version.

It basically came down to codecs throwing an exception on non-ascii characters (different kinds for py2 and py3)